sn_interface/src/network_knowledge/test_utils.rs

```diff
@@ -0,0 +1,283 @@
+    (0..count)
+        .map(|index| {
+            let age = if age_diff && index < count - 1 {
+                MIN_ADULT_AGE + 1
+            } else {
+                MIN_ADULT_AGE
+            };
+            NodeInfo::new(
+                crate::types::keys::ed25519::gen_keypair(&prefix.range_inclusive(), age),
+                gen_addr(),
+            )
+        })
+        .sorted_by_key(|node| node.name())
+        .collect()
+}
+
+/// Generate a random `SectionAuthorityProvider` for testing.
+///
+/// The total number of members in the section will be `elder_count` + `adult_count`. A lot of
+/// tests don't require adults in the section, so zero is an acceptable value for
+/// `adult_count`.
+///
+/// An optional `sk_threshold_size` can be passed to specify the threshold when the secret key
+/// set is generated for the section key. Some tests require a low threshold.
+pub fn random_sap_with_rng<R: RngCore>(
+    rng: &mut R,
+    prefix: Prefix,
+    elder_count: usize,
+    adult_count: usize,
+    sk_threshold_size: Option<usize>,
+) -> (SectionAuthorityProvider, Vec<NodeInfo>, bls::SecretKeySet) {
+    let nodes = gen_sorted_nodes(&prefix, elder_count + adult_count, false);
+    let elders = nodes.iter().map(NodeInfo::peer).take(elder_count);
+    let members = nodes.iter().map(|i| NodeState::joined(i.peer(), None));
+    let poly = bls::poly::Poly::random(sk_threshold_size.unwrap_or(0), rng);
+    let sks = bls::SecretKeySet::from(poly);
+    let section_auth = SectionAuthorityProvider::new(elders, prefix, members, sks.public_keys(), 0);
+    (section_auth, nodes, sks)
+}
+
+pub fn random_sap(
+    prefix: Prefix,
+    elder_count: usize,
+    adult_count: usize,
+    sk_threshold_size: Option<usize>,
+) -> (SectionAuthorityProvider, Vec<NodeInfo>, bls::SecretKeySet) {
+    random_sap_with_rng(
+        &mut rand::thread_rng(),
+        prefix,
+        elder_count,
+        adult_count,
+        sk_threshold_size,
+    )
+}
+
+/// Generate a random `SectionAuthorityProvider` for testing.
+///
+/// The same as `random_sap`, but instead the secret key is provided. This can be useful for
+/// creating a section to share the same genesis key as another one.
+pub fn random_sap_with_key(
+    prefix: Prefix,
+    elder_count: usize,
+    adult_count: usize,
+    sk_set: &bls::SecretKeySet,
+) -> (SectionAuthorityProvider, Vec<NodeInfo>) {
+    let nodes = gen_sorted_nodes(&prefix, elder_count + adult_count, false);
+    let elders = nodes.iter().map(NodeInfo::peer).take(elder_count);
+    let members = nodes.iter().map(|i| NodeState::joined(i.peer(), None));
+    let section_auth =
+        SectionAuthorityProvider::new(elders, prefix, members, sk_set.public_keys(), 0);
+    (section_auth, nodes)
+}
+
+// Create signature for the given payload using the given secret key.
+pub fn prove<T: Serialize>(secret_key: &bls::SecretKey, payload: &T) -> Result<KeyedSig> {
```

Comment:
Let's give this a more descriptive name?
```suggestion
pub fn keyed_sig<T: Serialize>(secret_key: &bls::SecretKey, payload: &T) -> Result<KeyedSig> {
```

Reply:
Just realized this function was just being moved. We can skip the rename as well